### PR TITLE
Copy previous definition when updating views.

### LINKED ISSUE #49: View generator should copy previous definition when it exists
Usually when editing a view you want to tweak what's already there. We should be copying the file rather than creating a blank one.

### PR DESCRIPTION
**Pull request by DamirSvrtan**
Should resolve #49.

### CODE REVIEW

File: lib/generators/scenic/view/view_generator.rb [hunk 1]
```diff
@@ -58,6 +58,16 @@ def migration_class_name
...
       private
 
+      def previous_definition_content
+        if creating? && previous_version > 0
+          File.read(previous_definition.path)
+        end
+      end
+
+      def previous_definition
+        @previous_definition ||= Scenic::Definition.new(plural_file_name, previous_version)
```
Comment (DamirSvrtan) [inline lib/generators/scenic/view/view_generator.rb]: Now I see that there isn't a need to memoize that value.

**Comment by derekprior** [comment]
Thanks for the contribution. Looks like there is a failing spec in your branch:

```

  1) Scenic::Generators::ViewGenerator updates an existing view
     Failure/Error: run_generator ["search"]
     Thor::Error:
       Could not find "/Users/derek/src/scenic/spec/dummy/db/views/searches_v01.sql" in any of your source paths. Your current source paths are:
       /Users/derek/src/scenic/lib/generators/scenic/view/templates
     # ./lib/generators/scenic/view/view_generator.rb:20:in `create_view_definition'
     # ./spec/generators/scenic/view/view_generator_spec.rb:21:in `block (2 levels) in <top (required)>'

Finished in 0.0946 seconds (files took 0.74655 seconds to load)
34 examples, 1 failure

Failed examples:

rspec ./spec/generators/scenic/view/view_generator_spec.rb:15 # Scenic::Generators::ViewGenerator updates an existing view
```

Would you mind having a look at it? I might not get a chance to dig into it for a bit. Also, this reminds me to enable travis...

**Comment by derekprior** [comment]
I think this failing spec had to do with a misconfiguration in travis. Taking another look at this PR.

**Comment by derekprior** [comment]
Closed in 28ef2dc0e815d41b77b15bbd79d9101991f3dfc1

**Comment by derekprior** [comment]
Thank you for the contribution

**Pull request closed without merging**